dist/redhat: prevent removing .mount files during upgrade

syuu1228 opened the pull request:
We removed %ghost for .mount files at a677c46, it was for prevent
removing .mount files during package remove.
However, after a677c46 rpm removes .mount files during package upgrade,
because rpm things removed %ghost files are now dropped from package.
To fix both problem, specify .mount files as "%ghost %config".
It will keep files both package upgrade and package remove.

Fixes #8924

syuu1228 commented:
Note that I didn't reverted a677c46, because Debian fix on the commit is correct, only RPM fix is wrong.

avikivity commented:
@syuu1228 because we'll want to backport this patch, I reverted a677c4667251f1b26bb28eff14625c64bc3c41bc (in 0d87744ba091c964f5ec251584b5356f63923c41), so #8959 can be backported cleanly. So please rebase on top of master once it promotes.

avikivity commented on dist/redhat/scylla.spec:
```diff
@@ -141,7 +141,9 @@ rm -rf $RPM_BUILD_ROOT
 /etc/systemd/system/scylla-server.service.d/dependencies.conf
+%ghost %config /etc/systemd/system/var-lib-systemd-coredump.mount
 %ghost /etc/systemd/system/scylla-cpupower.service
+%ghost %config /etc/systemd/system/var-lib-scylla.mount
```
I thought we'll need two separate lines, good to see it works in one line (it means the rpm maintainers considered this case)

syuu1228 replied:
@avikivity it just worked with one line, and also rpmlint does not causes error with this.
I could find one line examples on the web:
https://git.sailfishos.org/mer-core/rpm/blob/master/rpm/rpm.spec#L216

Also, it seems maintainers are thinking it's expected behavior:
https://bugzilla.redhat.com/show_bug.cgi?id=1516938
https://github.com/rpm-software-management/rpm/commit/233e4573d1d21a36d06874fea6c594428033802e

syuu1228 commented:
> @syuu1228 because we'll want to backport this patch, I reverted [a677c46](https://github.com/scylladb/scylla/commit/a677c4667251f1b26bb28eff14625c64bc3c41bc) (in [0d87744](https://github.com/scylladb/scylla/commit/0d87744ba091c964f5ec251584b5356f63923c41)), so #8959 can be backported cleanly. So please rebase on top of master once it promotes.

Rebased with latest next.
